### PR TITLE
Feature/joystick

### PR DESCRIPTION
**Pull request by TStrobino**
Prelim. final demo.

### CODE REVIEW

File: Aero390Spoilers/Aero390Spoilers/AircraftGUI.Designer.cs [hunk 1]
```diff
@@ -163,7 +165,7 @@ private void InitializeComponent()
             this.WowButton.FlatAppearance.MouseDownBackColor = System.Drawing.Color.DimGray;
             this.WowButton.FlatAppearance.MouseOverBackColor = System.Drawing.Color.Transparent;
             this.WowButton.FlatStyle = System.Windows.Forms.FlatStyle.Flat;
-            this.WowButton.Font = new System.Drawing.Font("B612", 9.749999F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
+            this.WowButton.Font = new System.Drawing.Font("Microsoft Sans Serif", 9.749999F, System.Drawing.FontStyle.Regular, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
```
Comment (Aeroponz) [inline Aero390Spoilers/Aero390Spoilers/AircraftGUI.Designer.cs]: Wrong font.

File: Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs [hunk 2]
```diff
@@ -235,46 +279,106 @@ private void PowerLossMalfunction(bool MalfStatus)
...
+            //        }
+            //        double DeflectionPercent = ((double)(GUIOwnship.SpoilerLeverPosition) / 10.0) * 100;
+            //        double FromDeflection = GUIOwnship.SpoilerDeflectionPercentage[0];
+            //        if (GUIOwnship.SpoilerLeverPosition <= 0) DeflectionPercent = 0;
+            //        if (temp <= 0) FromDeflection = 0;
+            //        Thread IncrementSpoilers = new Thread(() => RefreshSpoilerActuation((int)FromDeflection, (int)DeflectionPercent, !GUIOwnship.WeightOnWheels));
+            //        SpoilerThreadRunning = true;
+            //        IncrementSpoilers.Start();
+            //    }
+            //}
```
Comment (Aeroponz) [inline Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs]: Clean up required

File: Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs [hunk 3]
```diff
@@ -545,21 +649,21 @@ private void RefreshEICASSystemStatus()
...
+            //Spoiler5PGB.Value = 100 - GUIOwnship.SpoilerDeflectionPercentage[4];
             Spoiler5PGB.Refresh();
-            Spoiler6PGB.Value = 100 - GUIOwnship.SpoilerDeflectionPercentage[5];
+            //Spoiler6PGB.Value = 100 - GUIOwnship.SpoilerDeflectionPercentage[5];
             Spoiler6PGB.Refresh();
-            Spoiler7PGB.Value = 100 - GUIOwnship.SpoilerDeflectionPercentage[6];
+            //Spoiler7PGB.Value = 100 - GUIOwnship.SpoilerDeflectionPercentage[6];
             Spoiler7PGB.Refresh();
-            Spoiler8PGB.Value = 100 - GUIOwnship.SpoilerDeflectionPercentage[7];
+            //Spoiler8PGB.Value = 100 - GUIOwnship.SpoilerDeflectionPercentage[7];
```
Comment (Aeroponz) [inline Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs]: Clean up required

File: Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs [hunk 4]
```diff
@@ -662,9 +766,154 @@ private void RefreshSpoilerActuation(int CurrentDeflection, int TargetDeflection
                 CurrentDeflection += increment;
                 Thread.Sleep(30);
             }
-            SpoilerThreadRunning = false;
+            //SpoilerThreadRunning = false;
```
Comment (Aeroponz) [inline Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs]: clean up

File: Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs [hunk 5]
```diff
@@ -704,8 +953,8 @@ private void RepositionTo(string Reposition)
                         FlapLever.Value = 0;
                         GUIOwnship.FlapLeverPosition = 0;
                         GUIOwnship.GrossWeightLbs = 30000;
-                        LENGThrottle.Value = 8;
-                        RENGThrottle.Value = 8;
+                        //LENGThrottle.Value = 8;
+                        //RENGThrottle.Value = 8;
```
Comment (Aeroponz) [inline Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs]: Clean up

File: Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs [hunk 6]
```diff
@@ -722,7 +971,7 @@ private void RepositionTo(string Reposition)
                 case ("Approach"):
                     {
                         GUIOwnship.AltitudeASL = 1073 + GUIOwnship.RunwayAltASL;
-                        GUIOwnship.AoA = -3;
+                        //GUIOwnship.AoA = -3;
```
Comment (Aeroponz) [inline Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs]: Clean up

File: Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs [hunk 7]
```diff
@@ -735,18 +984,18 @@ private void RepositionTo(string Reposition)
                         GUIOwnship.RThrottlePosition = 4;
                         SpoilerLever.Value = 0;
                         GUIOwnship.SpoilerLeverPosition = 0;
-                        ControlWheelBar.Value = 0;
-                        GUIOwnship.SWControlWheelPosition = 0;
-                        GUIOwnship.VS = -600;
+                        //ControlWheelBar.Value = 0;
+                        //GUIOwnship.SWControlWheelPosition = 0;
+                        //GUIOwnship.VS = -600;
```
Comment (Aeroponz) [inline Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs]: Clean up

File: Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs [hunk 8]
```diff
@@ -766,11 +1015,11 @@ private void RADALTStub()
                 }
                 Thread.Sleep(100);
             }
-            GUIOwnship.VS = 0;
+            //GUIOwnship.VS = 0;
```
Comment (Aeroponz) [inline Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs:975]: Clean up

File: Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs [hunk 9]
```diff
@@ -998,5 +1252,14 @@ private void AppRepoButton_Click(object sender, EventArgs e)
...
 
+        private void RENGThrottle_Scroll(object sender, EventArgs e)
+        {
+
+        }
+
+        private void VerticalSpeedIndicatorInstrumentControl1_Click(object sender, EventArgs e)
+        {
+
+        }
```
Comment (Aeroponz) [inline Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs:1220]: Clean up. Functions are empty

File: Aero390Spoilers/Ownship/Ownship.cs [hunk 10]
```diff
@@ -203,39 +201,43 @@ public void RemoveEicasMessage(EICASMessage iMsg)
...
+        public int MFS_as_brake = 0;
         public Stack<EICASMessage> EICASMessages = new Stack<EICASMessage>(); //Lines 1 to 8 for CW Messages, 9 to 11 for status
 
         //WARNING SYSTEM
         public bool WarningActive { get; set; }
         public bool CautionActive { get; set; }
 
         //MALFUNCTIONS AND SWITCHES
-        public bool Switch1On { get; set; }
+        public bool Switch1On { get; set; } = true;
```
Comment (Aeroponz) [inline Aero390Spoilers/Ownship/Ownship.cs]: The initialization should not be done here. Please move it to the constructor.

File: Aero390Spoilers/Ownship/Ownship.cs [hunk 11]
```diff
@@ -203,39 +201,43 @@ public void RemoveEicasMessage(EICASMessage iMsg)
...
         public bool Switch2On { get; set; }
         public bool Switch3On { get; set; }
         public bool Switch4On { get; set; }
         public bool Switch5On { get; set; }
         public bool Switch6On { get; set; }
-        public bool MalfPwrLoss { get; set; }
+        public bool MalfPwrLoss { get; set; } = true;
```
Comment (Aeroponz) [inline Aero390Spoilers/Ownship/Ownship.cs]: The initialization should not be done here. Please move it to the constructor.

**Review by Aeroponz — CHANGES_REQUESTED** [review]

File: Aero390Spoilers/Aero390Spoilers/AircraftGUI.Designer.cs (hunk 1 above)
Comment (TStrobino) [reply Aero390Spoilers/Aero390Spoilers/AircraftGUI.Designer.cs]: This was auto-generated. The button is a static image, tho. I changed it back to B612, but saw no effect in the actual GUI.

File: Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs (hunk 2 above)
Comment (TStrobino) [reply Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs]: done.

File: Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs (hunk 3 above)
Comment (TStrobino) [reply Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs]: done.

File: Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs (hunk 9 above)
Comment (TStrobino) [reply Aero390Spoilers/Aero390Spoilers/AircraftGUI.cs:1220]: I'd personally leave them there for future reference. We may need them to actually do something when we remove joystick dependency.

File: Aero390Spoilers/Ownship/Ownship.cs (hunk 10 above)
Comment (TStrobino) [reply Aero390Spoilers/Ownship/Ownship.cs]: done!

**Pull request merged**